spec_macros/src/derive_spec/spec_struct.rs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2025 Elektrobit Automotive GmbH
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License, Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{FieldsNamed, Ident, Type, Visibility, parse_quote};
+
+use crate::utils::{
+    DerivedSpec, extract_inner, get_doc_attrs, get_prost_enum_type, get_prost_map_enum_value_type,
+    get_spec_field_attrs, has_mandatory_attr, inner_hashmap_type_path, inner_vec_type_path,
+    is_custom_type_path, is_option_type_path, to_spec_ident, to_spec_type, wrap_in_option,
+};
```

Comment:
Same logic with the imports
```suggestion
use crate::utils::{
    DerivedSpec, extract_inner, get_doc_attrs, get_prost_enum_type, get_prost_map_enum_value_type,
    get_spec_field_attrs, has_mandatory_attr, inner_hashmap_type_path, inner_vec_type_path,
    is_custom_type_path, is_option_type_path, to_spec_ident, to_spec_type, wrap_in_option,
};

use proc_macro2::TokenStream;
use quote::quote;
use syn::{FieldsNamed, Ident, Type, Visibility, parse_quote};
```